include/nbl/asset/IMeshBuffer.h

```diff
@@ -14,214 +14,281 @@ namespace asset
+        inline uint32_t getBindingNumForAttribute(uint32_t attrId) const
+        {
+            const auto* ppln = m_pipeline.get();
+            const auto& vtxInputParams = ppln->getVertexInputParams();
+            return vtxInputParams.attributes[attrId].binding;
+        }
+        inline E_FORMAT getAttribFormat(uint32_t attrId) const
+        {
+            const auto* ppln = m_pipeline.get();
+            const auto& vtxInputParams = ppln->getVertexInputParams();
+            return static_cast<E_FORMAT>(vtxInputParams.attributes[attrId].format);
+        }
+        inline uint32_t getAttribStride(uint32_t attrId) const
+        {
+            const auto* ppln = m_pipeline.get();
+            const auto& vtxInputParams = ppln->getVertexInputParams();
+            const uint32_t bnd = getBindingNumForAttribute(attrId);
+            return vtxInputParams.bindings[bnd].stride;
+        }
+        inline uint32_t getAttribOffset(uint32_t attrId) const
+        {
+            const auto* ppln = m_pipeline.get();
+            const auto& vtxInputParams = ppln->getVertexInputParams();
+            return vtxInputParams.attributes[attrId].relativeOffset;
+        }
+        inline const SBufferBinding<const BufferType>& getAttribBoundBuffer(uint32_t attrId) const
+        {
+            const uint32_t bnd = getBindingNumForAttribute(attrId);
+            return reinterpret_cast<const SBufferBinding<const BufferType>&>(m_vertexBufferBindings[bnd]);
+        }
+        inline uint64_t getAttribCombinedOffset(uint32_t attrId) const
+        {
+            const auto& buf = getAttribBoundBuffer(attrId);
+            return buf.offset+static_cast<uint64_t>(getAttribOffset(attrId));
+        }
+
+        //
+        inline const SBufferBinding<const BufferType>* getVertexBufferBindings() const
+        {
+            return reinterpret_cast<const SBufferBinding<const BufferType>*>(m_vertexBufferBindings);
+        }
+        inline const SBufferBinding<const BufferType>& getIndexBufferBinding() const
+        {
+            return reinterpret_cast<const SBufferBinding<const BufferType>&>(m_indexBufferBinding);
+        }
+
+        //!
+        inline const SBufferBinding<const BufferType>& getInverseBindPoseBufferBinding() const
+        {
+            return reinterpret_cast<const SBufferBinding<const BufferType>&>(m_inverseBindPoseBufferBinding);
+        }
+        //!
+        inline const SBufferBinding<const BufferType>& getJointAABBBufferBinding() const
+        {
+            return reinterpret_cast<const SBufferBinding<const BufferType>&>(m_jointAABBBufferBinding);
+        }
+        //!
+        inline const SkeletonType* getSkeleton() const
+        {
+            return m_skeleton.get();
+        }
+
+        //! Returns max joint influences
+        inline auto getMaxJointsPerVertex() const { return maxJointsPerVx; }
+
+        //!
+        virtual inline bool isSkinned() const
```

Comment:
you could also validate here if attribIds for skinning are set correctly

Reply:
I do that in `ICPUMeshBuffer` confer with the override

Reply:
I do it indirectly via a call to `deduceMaxJointsPerVertex`